### PR TITLE
Handle affine matrix mismatches better by exposing `-set-qform-to-sform`

### LINKED ISSUE #3249: 'mnc2nii' corrupts qform, but SCT recommends '-set-sform-to-qform'
more to come soon-- @mpompolas can chip in to document this issue

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've applied a [release milestone](https://github.com/neuropoly/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/neuropoly/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [X] I've assigned a reviewer

#### PR contents

- [X] I've consulted [SCT's internal developer documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [X] I've updated the [relevant documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

In #3249, we learned that our current recommendation (`-set-sform-to-qform`) is not sufficient to fix all affine mismatch issues. So, this PR makes the following changes:

* Exposes the complementary operation (`-set-qform-to-sform`)
* Broadens the error message so that users are encouraged to inspect the affine matrices and understand how to fix their own affine matrices.

### Future work

I also wanted to point users to a documentation page about what valid affines look like (so that users can learn how to spot issues). But, there's not really a good page to link! 

* Nibabel has pages that are close (["Coordinate systems and affines"](https://nipy.org/nibabel/coordinate_systems.html), ["The NIFTI affines"](https://nipy.org/nibabel/nifti_images.html#the-nifti-affines)) but those pages contain a lot of extraneous information that could confuse/overwhelm readers. 
* The NIFTI-1 spec also has some useful pages (["Docs on qform and sform"](https://nifti.nimh.nih.gov/nifti-1/documentation/nifti1fields/nifti1fields_pages/qsform.html)), but these are written in a very technical way and don't seem friendly to non-developers.

So, if others feel it is worth the effort, I could also write a quick SCT documentation page on how to inspect the output of `sct_image -header`. But, this PR could also be sufficient on its own.

## Linked issues

Fixes #3249.
Depends on #3317.

### CODE REVIEW

**Review by jcohenadad — APPROVED** [review]
LGTM!

**Comment by joshuacwnewton** [comment]
Thanks for the review/approval, @jcohenadad! 

As a heads up, I'm going to wait until #3317 is merged to merge this PR (since the message is tied to `sct_image -header`). But, I'll keep an eye on both so that 5.3.0 can be released ASAP. :slightly_smiling_face:

**Pull request merged**
